resource-definitions/k8s-cluster-git/runtime/github-for-gitops.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: entity.humanitec.io/v1b1
+kind: Definition
+metadata:
+  id: github-for-gitops
+entity:
+  name: github-for-gitops
+  driver_type: humanitec/k8s-cluster-git
+  type: k8s-cluster
+  driver_inputs:
+    values:
+      # Git repository for pushing manifests
+      url: git@github.com:example-org/gitops-repo.git
+      # Branch in the git repository, optional. If not specified, the default branch is used.
+      branch: development
+      # Path in the git repository, optional. If not specified, the root is used.
+      path: "${context.app.id}/${context.env.id}"
+      # Load Balancer, optional. Though it's not related to the git, it's used to create ingress in the target K8s cluster.
+      loadbalancer: 35.10.10.10
+    secrets:
+      credentials:
+        ssh_key: my-git-ssh-key
+        # Alternative to ssh_key: password or Personal Account Token
+        # password: my-git-ssh-pat
+  
+  # To co-provision another k8s-cluster resource from which the Orchestrator will fetch runtime info
```

Comment:
Both cluster resources, the present one the co-provisioned one, are of type `k8s-cluster`.

What is the mechanism that makes the Platform Orchestrator tap into this co-provisioned cluster resource for runtime data, and not the present one?

Reply:
It works because the co-provision key includes the resource id (`k8s-cluster-runtime`).

Reply:
Pushed a change that should clarify this bit.

Reply:
The change explains better how the matching works, that sure helps.

I was more getting at the runtime setup though. In my Resource Graph, I will have two Active Resources of type `k8s-cluster`:

- One with class `default` that is implicitly matched, and provisioned using the Resource Definition in `github-for-gitops.yaml`, of `driver_type: humanitec/k8s-cluster-git`
- One with class `k8s-cluster-runtime` that was co-provisioned, using the Resource Definition in `gke-temporary-credentials.yaml`, of  `driver_type: humanitec/k8s-cluster-gke`

Normally, you would only have one `k8s-cluster` Resource in your Graph, so that's also the runtime cluster.

Now how does the Platform Orchestrator determine which of the two Active Resources to use for runtime data, when there is more than one?

Or in other words, what is the key property that defines the "runtime cluster"?

Reply:
The two `k8s-cluster` Resource Definitions are not using a different class but a different Resource Id.

> Now how does the Platform Orchestrator determine which of the two Active Resources to use for runtime data, when there is more than one?

The Platform Orchestrator looks for a `k8s-cluster` with Resource Id `k8s-cluster-runtime` to fetch runtime information, if it does not find it, defaults to `k8s-cluster` with Resource Id `k8s-cluster`.
I'll try to explain this here, or do you think I should do it in another place?

Reply:
I understand this is kind of hidden magic and this is exactly what I'd like to resolve / explain with this piece of docs.

Reply:
Yes it's `id`, not `class`, my bad 🙈

This explanation would be perfect just to add to the README if this example as is.

It this also the logic used for the [`runtime` API endpoint](https://api-docs.humanitec.com/#tag/RuntimeInfo)? If so, I suggest adding it there so we have it written down right at the source.

For the docs, we have tickets in the backlog to describe GitOps setup. I added a hint to integrate the example, so nothing to do now.

Reply:
Ok, all good then, I'll add this stuff to the README.

Very good point about adding this info to the [runtime API endpoint](https://api-docs.humanitec.com/#tag/RuntimeInfo/operation/getRuntime), I'll do it. Thank you!

Reply:
Great explanation in the README, thanks a lot! 🎉

Reply:
Thank you Tobi!